update role for query insights

ansjcy opened the pull request:
### Description
* Category: Maintenance
* Why these changes are required: We need to update the role for query insights access to allow addition APIs we introduced in 3.x

### Issues Resolved
[List any issues this PR will resolve]

Is this a backport? If so, please add backport PR # and/or commits #, and remove `backport-failed` label from the original PR.

Do these changes introduce new permission(s) to be displayed in the static dropdown on the front-end? If so, please open a draft PR in the security dashboards plugin and link the draft PR here

### Testing
[Please provide details of testing done: unit testing, integration testing and manual testing]

### Check List
- [ ] New functionality includes testing
- [ ] New functionality has been documented
- [ ] New Roles/Permissions have a corresponding security dashboards plugin PR
- [ ] API changes companion pull request [created](https://github.com/opensearch-project/opensearch-api-specification/blob/main/DEVELOPER_GUIDE.md)
- [ ] Commits are signed per the DCO using --signoff

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/security/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

cwperks reviewed (approved):
Approving but can you please explain the change inded pattern here? Would any users need to perform a migration from the older naming to newer naming?

ansjcy commented:
There are 2 changes in this PR:
1. We added permission to access live queries API
2. Fixed the index permission to the right one. There's no top_queries_by index, It's a bug that could cause users not able to access the correct top_queries-* index. So there's no migration needed.

shikharj05 commented on config/roles.yml:
```diff
@@ -451,10 +451,11 @@ flow_framework_read_access:
     - 'cluster:admin/opensearch/insights/top_queries/*'
   index_permissions:
     - index_patterns:
-        - 'top_queries_by_*'
```
this was never used?

ansjcy replied:
yes, it should be `-`. This is defined in query insights plugin.

shikharj05 commented:
can you also add a change-log entry?

cwperks commented on config/roles.yml:
```diff
@@ -451,10 +451,11 @@ flow_framework_read_access:
   index_permissions:
     - index_patterns:
-        - 'top_queries_by_*'
+        - 'top_queries-*'
```
+1 to @DarshitChanpura 's comment above. Should we mix _ and - here?

other question I had is whether these indices should be directly queryable by a user? If these are not system indices then any user with requisite permissions on these indices can write to them directly or delete them. Can the query insights plugin handle such cases gracefully?

ansjcy replied:
> should be directly queryable by a user

When we design this feature we want it to be similar to audit log and it should be queryable by a user. So this is expected.
